cmd/autoheal/alerts_worker.go

```diff
@@ -120,94 +128,77 @@ func (h *Healer) cancelHealing(alert *alertmanager.Alert) error {
 	return nil
 }
 
-func (h *Healer) checkConditions(rule *monitoring.HealingRule, alert *alertmanager.Alert) bool {
+func (h *Healer) checkRule(rule *monitoring.HealingRule, alert *alertmanager.Alert) (matches bool, err error) {
 	glog.Infof(
-		"Checking conditions of rule '%s' for alert '%s'",
+		"Checking rule '%s' for alert '%s'",
 		rule.ObjectMeta.Name,
 		alert.Name(),
 	)
-	if rule.Spec.Conditions != nil && len(rule.Spec.Conditions) > 0 {
-		for i := 0; i < len(rule.Spec.Conditions); i++ {
-			if !h.checkCondition(&rule.Spec.Conditions[i], alert) {
-				return false
-			}
-		}
+	matches, err = h.checkMap(alert.Labels, rule.Labels)
+	if !matches || err != nil {
+		return
 	}
-	return true
-}
-
-func (h *Healer) checkCondition(condition *monitoring.HealingCondition, alert *alertmanager.Alert) bool {
-	matched, err := regexp.MatchString(condition.Alert, alert.Name())
-	if err != nil {
-		glog.Errorf(
-			"Error while checking if alert name '%s' matches pattern '%s': %s",
-			alert.Name(),
-			condition.Alert,
-			err.Error(),
-		)
-		matched = false
+	matches, err = h.checkMap(alert.Annotations, rule.Annotations)
+	if !matches || err != nil {
+		return
 	}
-	return matched
+	return
 }
 
-func (h *Healer) runActions(rule *monitoring.HealingRule, alert *alertmanager.Alert) error {
-	if rule.Spec.Actions != nil && len(rule.Spec.Actions) > 0 {
-		glog.Infof(
-			"Running actions of healing rule '%s' for alert '%s'",
-			rule.ObjectMeta.Name,
-			alert.Name(),
-		)
-		for i := 0; i < len(rule.Spec.Actions); i++ {
-			err := h.runAction(rule, &rule.Spec.Actions[i], alert)
-			if err != nil {
-				return err
+func (h *Healer) checkMap(values, patterns map[string]string) (result bool, err error) {
+	if len(patterns) > 0 {
```

Comment:
nit: I would prefer if result was explicitly set to false at this point for readability (if `len(patterns) > 0`)

That might be a personal preference though

Reply:
Sure, I am doing that change, then I am merging.

Reply:
Ah, you already merged the pull request, thanks :-) !